tests/pause_workflow_execution_test.go

```diff
@@ -0,0 +1,179 @@
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/api/workflowservice/v1"
+	sdkclient "go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/tests/testcore"
+)
+
+type PauseWorkflowExecutionSuite struct {
+	testcore.FunctionalTestBase
+}
+
+func TestPauseWorkflowExecutionSuite(t *testing.T) {
+	s := new(PauseWorkflowExecutionSuite)
+	suite.Run(t, s)
+}
+
+func (s *PauseWorkflowExecutionSuite) SetupTest() {
+	s.FunctionalTestBase.SetupTest()
+	s.OverrideDynamicConfig(dynamicconfig.WorkflowPauseEnabled, true)
+}
+
+func (s *PauseWorkflowExecutionSuite) TestPauseWorkflowExecution() {
+	const (
+		testEndSignal = "test-end"
+		pauseIdentity = "functional-test"
+		pauseReason   = "pausing workflow for acceptance test"
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	workflowFn := func(ctx workflow.Context) (string, error) {
+		signalCh := workflow.GetSignalChannel(ctx, testEndSignal)
+		var signalPayload string
+		signalCh.Receive(ctx, &signalPayload)
+		return signalPayload, nil
+	}
```

Comment:
`workflowFn` is duplicated, could also break this out

Reply:
will do.